### PR TITLE
do not use deprecated features by Elixir version

### PR DESCRIPTION
**Pull request by chulkilee**
Deprecation

- Elixir 1.14 - `use Bitwise` is deprecated
- Elixir 1.10 - `Application.compile_env/3` is introduced

Also add Elixir 1.14.1 & OTP 25.1.2 to CI

### CODE REVIEW

File: lib/myxql/protocol/auth.ex [hunk 1]
```diff
@@ -1,7 +1,7 @@
 defmodule MyXQL.Protocol.Auth do
   @moduledoc false
 
-  # TODO: remove when we require Elixir v1.10+
+  # TODO: remove when we require Elixir v1.14+
```
Comment (wojtekmach) [inline lib/myxql/protocol/auth.ex:4]: the comment is wrong, can you remove it? It is fine that we call `require Bitwise; Bitwise.bxor`.

Comment (chulkilee) [reply lib/myxql/protocol/auth.ex:4]: It is fine but we can remove `require Bitwise` hereon Elixir 1.14 as it's  [deprecated](https://github.com/elixir-lang/elixir/blob/v1.14/CHANGELOG.md) and this file does not use operator (which requires `import Bitwise` like other modules)

But Elixir <1.14 needs `require` here.

Comment (wojtekmach) [reply lib/myxql/protocol/auth.ex:4]: sorry, the comment was correct after all. The change between v1.9 and v1.10 is basically:

```diff
-  defmacro bxor(left, right) do
+  def bxor(left, right) do
```

that's why we previously needed a require and nowadays we don't.

Comment (wojtekmach) [inline lib/myxql/protocol/auth.ex:4]:
```suggestion
  # TODO: remove when we require Elixir v1.10+
```

**Pull request merged**